Assets/Shaders/Caustics.gdshader

```diff
@@ -111,29 +123,50 @@ mat3 makeRotationDir(vec3 direction, vec3 up){
 }
 
 void light(){
-	// Mapping texture
-	vec3 world_light = (vec4(LIGHT, 1.0) * INV_CAMERA_MATRIX).xyz;
-	mat3 lrm = makeRotationDir(world_light, vec3(0.0, 1.0, 0.0));
-	mat4 t = mat4(lrm);
-	t[3].xyz = positionWS;
-	mediump vec2 uv = (t * vec4(positionWS, 1.0)).xy;
-	mediump vec2 uv1 = panner(uv, 0.75 * caustics_speed, 1.0 / caustics_scale);
-	mediump vec2 uv2 = panner(uv, caustics_speed, -1.0 / caustics_scale);
-	
-	mediump vec3 caus1 = aberration_sample(caustics_noise, uv1, chroma_split);
-	mediump vec3 caus2 = aberration_sample(caustics_noise, uv2, chroma_split);
-	caus1.rgb = pow(caus1.rgb, vec3(caustics_power));
-	caus2.rgb = pow(caus2.rgb, vec3(caustics_power));
-	
-	mediump vec3 c = min(caus1.rgb, caus2.rgb) * caustics_strength;
-	
-	// Luminance Fade
-	mediump float scene_luma = luminance(screen_color);
-	scene_luma = texture(luma_gradient, vec2(scene_luma)).r;
-	mediump float luminanceMask = mix(1.0, scene_luma, caustics_luminance_mask_strength);
-	
-	// Edge Fade
-	mediump float edgeFadeMask = 1.0 - clamp((distance(positionOS, vec3(0.0)) - caustics_fade_radius) / (1.0 - caustics_fade_strength), 0.0, 1.0);
-	
-	DIFFUSE_LIGHT += c * bbox_mask * luminanceMask * edgeFadeMask;
+	if (!LIGHT_IS_DIRECTIONAL) {
+		// Only process directional lights!
+	}
+	else 
+	{
+		// Mapping texture
+		// Transform LIGHT from view space to world space.
+		vec3 world_light = (INV_VIEW_MATRIX * vec4(LIGHT, 0.0)).xyz;
+		
+		mat3 lrm = makeRotationDir(world_light, vec3(0.0, 1.0, 0.0));
+		mat4 t = mat4(lrm);
+		
+		mediump vec2 uv = (t * vec4(positionWS, 1.0)).xy;
+		mediump vec2 uv1 = panner(uv, 0.75 * caustics_speed, 1.0 / caustics_scale);
+		mediump vec2 uv2 = panner(uv, caustics_speed, -1.0 / caustics_scale);
+		
+		mediump vec3 caus1 = aberration_sample(caustics_noise, uv1, chroma_split);
+		mediump vec3 caus2 = aberration_sample(caustics_noise, uv2, chroma_split);
+		caus1.rgb = pow(caus1.rgb, vec3(caustics_power));
+		caus2.rgb = pow(caus2.rgb, vec3(caustics_power));
+		
+		mediump vec3 c = min(caus1.rgb, caus2.rgb) * caustics_strength;
+		
+		// Luminance Fade
+		mediump float scene_luma = luminance(screen_color);
+		scene_luma = texture(luma_gradient, vec2(scene_luma)).r;
+		mediump float luminanceMask = mix(1.0, scene_luma, caustics_luminance_mask_strength);
+		
+		// Edge Fade
+		mediump float edgeFadeMask = 1.0 - clamp((distance(positionOS, vec3(0.0)) - caustics_fade_radius) / (1.0 - caustics_fade_strength), 0.0, 1.0);
+		
+		// Add this light's contribution to the caustics 
+		if (OUTPUT_IS_SRGB) { // Compatibility.
+			DIFFUSE_LIGHT += c * bbox_mask * luminanceMask * edgeFadeMask;
+		} else { // Forward_ and Mobile.
+			DIFFUSE_LIGHT += c * bbox_mask * luminanceMask * edgeFadeMask;
+		}
+		
+		// Hack to avoid a faint outline of the bounding box appearing.
+		// If the total diffuse light is very low, set alpha to 0.0 so nothing renders.
+		if (OUTPUT_IS_SRGB) { // Compatibility.
+			ALPHA = smoothstep(.0005, .001, DIFFUSE_LIGHT.g);
```

Comment:
Keying off of only the green (or red, or blue) value might be wrong here, since the chromatic aberration splits colors. It still looks generally okay to me, though. This could probably key off of some composite value like the average or minimum of the channels.

The smoothstep values are fairly arbitrary, just done to avoid a harsh step().